samples/sample_multi_transcode/src/pipeline_transcode.cpp

```diff
@@ -4568,38 +4568,45 @@ void DecreaseReference(mfxFrameData *ptr)
 }
 
 SafetySurfaceBuffer::SafetySurfaceBuffer(SafetySurfaceBuffer *pNext)
-    :m_pNext(pNext),
-     m_IsBufferingAllowed(true)
+    : m_pNext(pNext)
+    , m_IsBufferingAllowed(true)
+    , m_bRelEventHappened(false)
+    , m_bInsEventHappened(false)
 {
-    mfxStatus sts=MFX_ERR_NONE;
-    pRelEvent = new MSDKEvent(sts,false,false);
-    MSDK_CHECK_POINTER_NO_RET(pRelEvent);
-
-    pInsEvent = new MSDKEvent(sts,false,false);
-    MSDK_CHECK_POINTER_NO_RET(pInsEvent);
-
-} // SafetySurfaceBuffer::SafetySurfaceBuffer
+}
 
 SafetySurfaceBuffer::~SafetySurfaceBuffer()
 {
-    delete pRelEvent;
-    delete pInsEvent;
-} //SafetySurfaceBuffer::~SafetySurfaceBuffer()
+}
 
 mfxU32 SafetySurfaceBuffer::GetLength()
 {
     std::lock_guard<std::mutex> guard(m_mutex);
     return (mfxU32)m_SList.size();
 }
 
-mfxStatus SafetySurfaceBuffer::WaitForSurfaceRelease(mfxU32 msec)
+void SafetySurfaceBuffer::WaitForSurfaceRelease(mfxU32 msec)
 {
-    return pRelEvent->TimedWait(msec);
+    using namespace std::chrono;
+
+    std::unique_lock<std::mutex> lock(m_mutex);
+    if (m_RelEvent.wait_until(lock, system_clock::now() + milliseconds(msec), [this](){ return m_bRelEventHappened; }))
+        m_bRelEventHappened = false;
```

Comment:
@mgonchar,

Can we use `m_SList.size() == m_AsyncDepth` as a condition instead of introducing a new state flag m_bRelEventHappened?